src/Microsoft.AspNet.Mvc.ModelBinding/Internal/ModelBindingHelper.cs

```diff
@@ -100,7 +100,20 @@ internal static void AddModelErrorBasedOnExceptionType(ModelBindingContext bindi
         {
             if (IsFormatException(ex))
             {
-                bindingContext.ModelState.TryAddModelError(bindingContext.ModelName, ex.Message);
+                var modelState = bindingContext.ModelState[bindingContext.ModelName];
+                if (modelState == null)
+                {
+                    bindingContext.ModelState.TryAddModelError(bindingContext.ModelName, ex.Message);
+                }
+                else
```

Comment:
Eg. `ByteArrayModelBinder` does not set the `ModelState` with the `ValueProviderResult`. In this case `bindingContext.ModelState[bindingContext.ModelName]` is null.